src/questdb/ingress.pyx

```diff
@@ -1195,305 +1385,767 @@ cdef class Buffer:
+            f'not {auto_flush_bytes}')
+
+    if auto_flush_interval is not False and auto_flush_interval < 1:
+        raise ValueError(
+            '"auto_flush_interval" must be >= 1, '
+            f'not {auto_flush_interval}')
+
+    # Parse individual auto_flush parameters to C struct.
+    c_auto_flush.enabled = auto_flush
+
+    if auto_flush_rows is False:
+        c_auto_flush.row_count = -1
+    else:
+        c_auto_flush.row_count = auto_flush_rows
 
-    The ``Sender(.., auth=..)`` argument is a tuple of ``(kid, d, x, y)`` as
-    documented on the `QuestDB ILP authentication
-    <https://questdb.io/docs/reference/api/ilp/authenticate>`_ documentation.
-    Authentication is optional and disabled by default.
+    if auto_flush_bytes is False:
+        c_auto_flush.byte_count = -1
+    else:
+        c_auto_flush.byte_count = auto_flush_bytes
 
-    The ``Sender(.., tls=..)`` argument is one of:
+    if auto_flush_interval is False:
+        c_auto_flush.interval = -1
+    else:
+        c_auto_flush.interval = auto_flush_interval
 
-    * ``False``: No TLS encryption (default).
 
-    * ``True``: TLS encryption, accepting all common certificates as recognized
-      by either the `webpki-roots <https://crates.io/crates/webpki-roots>`_ Rust
-      crate (which in turn relies on https://mkcert.org/), or the OS-provided
-      certificate store.
+class TaggedEnum(Enum):
+    """
+    Base class for tagged enums.
+    """
 
-    * A ``str`` or ``pathlib.Path``: Path to a PEM-encoded certificate authority
-      file. This is useful for testing with self-signed certificates.
+    @property
+    def tag(self):
+        """
+        Short name.
+        """
+        return self.value[0]
 
-    * The special ``'os_roots'`` string: Use the OS-provided certificate store.
+    @property
+    def c_value(self):
+        return self.value[1]
 
-    * The special ``'webpki_roots'`` string: Use the `webpki-roots
-      <https://crates.io/crates/webpki-roots>`_ Rust crate to recognize
-      certificates.
+    @classmethod
+    def parse(cls, tag):
+        """
+        Parse from the tag name.
+        """
+        if tag is None:
+            return None
+        elif isinstance(tag, str):
+            for entry in cls:
+                if entry.tag == tag:
+                    return entry
+        elif isinstance(tag, cls):
+            return tag
+        else:
+            raise ValueError(f'Invalid value for {cls.__name__}: {tag!r}')
 
-    * The special ``'webpki_and_os_roots'`` string: Use both the `webpki-roots
-      <https://crates.io/crates/webpki-roots>`_ Rust crate and the OS-provided
-      certificate store to recognize certificates. (equivalent to `True`).
 
-    * The special ``'insecure_skip_verify'`` string: Dangerously disable all
-      TLS certificate verification (do *NOT* use in production environments).
+class Protocol(TaggedEnum):
+    """
+    Protocol to use for sending data to QuestDB.
```

Comment:
Add a line here that explains that HTTP should be preferred when available and that it provides additional transactions and error feedback support. Then link to the right :ref:`..` sections of the docs.